raft/README.md

```diff
@@ -140,11 +140,14 @@ The total state machine handling loop will look something like this:
     case <-s.Ticker:
       n.Tick()
     case rd := <-s.Node.Ready():
-      saveToStorage(rd.State, rd.Entries, rd.Snapshot)
```

Comment:
saveToStorage == persist + append. process Snapshot also should include storage.ApplySnapshot.

Not sure we should break everything out. I keep the func highlevel, since I do not want users to copy exactly what we have here.

Reply:
Hi, @xiang90 thanks for your comment. However, the main purpose of the PR is to reflect the first two user's responsibilities as described above "Write Entries, HardState and Snapshot to persistent storage in order,..." and "Second, all persisted log entries must be made available via an implementation of the Storage interface."

As a new user, when reading the README, I was struggling to match these two user's responsibilities and the sample pseudo-code. Therefore, I think it is clearer to separate persisting to storage and making persisted entries via the Storage interface.